### PR TITLE
Add ledger support to masternodes

### PR DESCRIPTION
**Pull request by Duddino**
## Abstract

Add ledger support to masternodes

## What does this PR address?

Masternodes were not be able to be controlled using a ledger.

## What features or improvements were added?

Ledger devices will now work using masternodes

## How does this benefit users?

Ledgers can now be used as a secure way to store the masternode collateral

### CODE REVIEW

File: scripts/wallet.js [hunk 1]
```diff
@@ -876,6 +881,7 @@ async function getHardwareWalletKeys(
             return cPubKey.publicKey;
         }
     } catch (e) {
+        console.log(e);
```
Comment (JSKitty) [inline scripts/wallet.js:884]: Nuke :bomb: 
```suggestion
```

File: scripts/masternode.js [hunk 2]
```diff
@@ -193,6 +206,22 @@ export default class Masternode {
         return [v + 27, ...signature];
     }
 
+    async getWalletPublicKey() {
+        if (masterKey.isHardwareWallet) {
+            const hi = await masterKey.getPublicKey(this.walletPrivateKeyPath);
+            console.log(hi);
+            return hexToBytes(hi);
```
Comment (JSKitty) [inline scripts/masternode.js:213]: Nuke the `hi` variable which looks like it was only used for logging/debugging purposes.
```suggestion
           return hexToBytes(await masterKey.getPublicKey(this.walletPrivateKeyPath));
```

File: scripts/masternode.js [hunk 3]
```diff
@@ -153,25 +154,37 @@ export default class Masternode {
...
-        );
-        return [v + 31, ...signature];
+        });
+
+        if (masterKey.isHardwareWallet) {
+            const { r, s, v } = await cHardwareWallet.signMessage(
+                this.walletPrivateKeyPath,
+                bytesToHex(toSign)
+            );
+            console.log(r, s, v);
```
Comment (JSKitty) [inline scripts/masternode.js:169]: Nuke :bomb:
```suggestion
```

File: scripts/masternode.js [hunk 4]
```diff
@@ -109,21 +114,17 @@ export default class Masternode {
...
-        );
         const mnPublicKey = hexToBytes(
             deriveAddress({
                 pkBytes: parseWIF(mnPrivateKey, true),
                 output: 'UNCOMPRESSED_HEX',
             })
         );
 
+        console.log(publicKey);
+
```
Comment (JSKitty) [inline scripts/masternode.js:127]: Nuke :bomb:
```suggestion
```

**Review by JSKitty — COMMENTED** [review]
A couple logs and nits suggested, otherwise, looking good at code-level!

**Review by vampyren — APPROVED** [review]
Have tested to setup a new MN on Allnode and added to MPW using key. Also voted and it works fine.

**Pull request merged**